source/postprocess/visualization.cc

```diff
@@ -446,30 +446,30 @@ namespace aspect
       if (output_format=="hdf5")
         {
           XDMFEntry new_xdmf_entry;
-          std::string     h5_solution_file_name = this->get_output_directory() + "solution/" + solution_file_prefix + ".h5";
-          std::string     xdmf_filename = this->get_output_directory() + "solution.xdmf";
+          const std::string h5_solution_file_name = "solution/" + solution_file_prefix + ".h5";
```

Comment:
The real fix is down in line 468. Previously we added the "solution/" string twice for the xdmf entry (one time in this line and one time down there). I just sorted creating the filenames around to make it consistent that: 

- None of the "*filename" variables contain the output_directory anymore
- All of them contain the "solution/" prefix already (if necessary)